cmd/swarmctl/service/flagparser/restart.go

```diff
@@ -10,25 +10,26 @@ import (
 )
 
 func parseRestart(flags *pflag.FlagSet, spec *api.ServiceSpec) error {
-	// parse restart-condition
-	condition, err := flags.GetString("restart-condition")
-	if err != nil {
-		return err
-	}
+	if spec.Task.Restart == nil || flags.Changed("restart-condition") {
```

Comment:
It can be written like this. 

```
if spec.Task.Restart == nil {
            spec.Task.Restart = &api.RestartPolicy{
                      Condition: api.RestartOnAny,
          }  
}            
```